### PR TITLE
[KOGITO-4110] - Single namespaced operator reacts on CRs created in d…

### PR DESCRIPTION
**Pull request by Kaitou786**
…ifferent namespaces

See: https://issues.redhat.com/browse/KOGITO-4110

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by Kaitou786** [comment]
The issue should also be available in 1.1.0 when we release, not sure if we should cherry-pick this or maybe mark it as `known-issues` in 1.1.0.

**Review by radtriste — APPROVED** [review]
Tested:

- Operator as cluster-wide => OK
- Operator as namespace-wide and kogito resource in same namespace => OK
- Operator as namespace-wide and kogito resource in a different namespace => OK, nothing created

**Comment by radtriste** [comment]
@Kaitou786 I guess the change would be easy to cherry-pick to 1.1.x. 
Kogito 1.1.0 will be released once Quarkus 1.11 is available so I guess we have some days to test this. and the whole weekend for nightly to run so there is any problem, we can still revert on Monday.
wdyt ?

**Comment by Kaitou786** [comment]
Yeah, that sounds good to me. Let me cherry-pick it to 1.1.x :)

File: main.go [hunk 1]
```diff
@@ -140,10 +140,12 @@ func getWatchNamespace() string {
...
 	var watchNamespaceEnvVar = "WATCH_NAMESPACE"
 
-	ns, found := os.LookupEnv(watchNamespaceEnvVar)
-	if !found {
-		setupLog.Info("unable to get WatchNamespace, "+
-			"the manager will watch and manage resources in all namespaces",
+	ns, _ := os.LookupEnv(watchNamespaceEnvVar)
+
+	// Check if operator is running as cluster scoped
+	if ns == "" {
```
Comment (spolti) [inline main.go:146]: I think that

```
len(ns) > 0
```

is better to verify empty vars.

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by radtriste** [comment]
@Kaitou786 You should rebase so that it works on AWS (because of `buildah`not supported anymore by Quarkus ...)

**Review by sutaakar — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (Kaitou786) [reply main.go]: Changed

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
The error is not related to this PR. @Kaitou786 @sutaakar is there a workaround for this DMN error?

**Comment by radtriste** [comment]
the error is almost resolved. https://github.com/kiegroup/kogito-cloud-operator/pull/714 should be merged once ok

**Review by spolti — APPROVED** [review]

**Pull request merged**